modules/imgproc/src/resize.avx2.cpp

```diff
@@ -132,7 +131,7 @@ class resizeNNInvokerAVX4 CV_FINAL :
     const Mat& src;
     Mat& dst;
     int* x_ofs;
-    double ify;
+    float ify;
```

Comment:
why changed?

Reply:
ocl and cuda use float in kernel, so the small error of floating (float/double) number (especially near x.5) will cause choosing different pixels when nearest resizing.